crates/biome_js_analyze/src/lint/nursery/use_exhaustive_switch_cases.rs

```diff
@@ -0,0 +1,314 @@
+use biome_analyze::{
+    FixKind, Rule, RuleDiagnostic, RuleDomain, context::RuleContext, declare_lint_rule,
+};
+use biome_console::markup;
+use biome_js_factory::make;
+use biome_js_syntax::{
+    AnyJsCallArgument, AnyJsExpression, AnyJsStatement, AnyJsSwitchClause, JsSwitchStatement, T,
+};
+use biome_js_type_info::{Literal, Type, TypeInner};
+use biome_rowan::{AstNode, AstNodeList, BatchMutationExt, TriviaPieceKind};
+
+use crate::JsRuleAction;
+use crate::services::typed::Typed;
+
+declare_lint_rule! {
+    /// Require switch-case statements to exhaustive.
+    ///
+    /// When working with union types or enums in TypeScript, it's common to want to write a switch
+    /// statement intended to contain a case for each possible type in the union or the enum.
```

Comment:
```suggestion
    /// When working with union types or enums in TypeScript, it's common to want to write a switch
    /// statement intended to contain a case for each possible variant.
```

[imho] The words `type` and `enum` appear close together, so the sentence feels a bit repetitive.  
You might consider replacing them with the single term `variant`.

Reply:
Actually I copied the description from typescript-eslint. But yeah, you are right, I think saying variant is enough here, too.